internal/query/parse.go

```diff
@@ -0,0 +1,37 @@
+package query
+
+import (
+	"time"
+
+	"github.com/bitcoin-sv/spv-wallet/models/filter"
+	"github.com/gin-gonic/gin"
+	"github.com/mitchellh/mapstructure"
+)
+
+func ParseSearchParams[T any](c *gin.Context, _ T) (*filter.SearchParams[T], error) {
```

Comment:
I suppose you've made the `_ T` argument to allow type deduction for generics 
- ...but this solution forces to create "unnecessary" instance of desired type to just get it's type. 
- so it affects runtime 

Consider a solution where a programmer explicitly provides the type in `[]`:
```golang
params, err := ParseSearchParams[ExampleConditionsForTests](c)

//instead of
params, err := ParseSearchParams(c, ExampleConditionsForTests{})
```